### PR TITLE
Externalise encryption key to a game.project setting

### LINKED ISSUE #5355: Move and externalise encryption key
**Task (REQUIRED):**
The encryption key is stored in ArchiveBuilder.java. Make it configurable and move it to a GitHub secret (and rotate it)

**Expected outcome (REQUIRED):**
The encryption key should change over time. Perhaps also allow the developer to chose a key.

https://forum.defold.com/t/unfold-wip-defold-modding-tool/66855/9?u=britzl

### PR DESCRIPTION
**Pull request by britzl**
Fixes #5355

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java [hunk 1]
```diff
@@ -200,7 +202,8 @@ private void createArchive(Collection<String> resources, RandomAccessFile archiv
             }
         }
 
-        ArchiveBuilder archiveBuilder = new ArchiveBuilder(root, manifestBuilder, use_vanilla_lua ? false : true, resourcePadding);
+        ArchiveBuilder archiveBuilder = new ArchiveBuilder(root, manifestBuilder, encrypt, resourcePadding);
+        archiveBuilder.setEncryptionKey(encryptionKey);
```
[the remark this answers is not in the document]
Comment (britzl) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java:206]: Oh :-) Good point. Converting this to a draft PR again.

[the remark this answers is not in the document]
Comment (britzl) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java:206]: In this case it has to remain in game.project and be read on startup by resource_archive.cpp

[the remark this answers is not in the document]
Comment (AGulev) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java:206]: I think it is not a secure option, especially if we take into account that game.projectc is available as a a plain text file.

Maybe we should use it as a trigger for using extender and some way to include it into the engine binary (NE where the engine checks in public header file or something like that, I dunno)
Ideally in the furor game.projectc should be encoded using the same key. 
Not sure if we have something like this for now

**Comment by britzl** [comment]
This feature needs further discussion. Refer to the discussion in the issue.

**Comment by AGulev** [comment]
maybe it's possible to do something like that but on Extender? https://github.com/defold/defold/blob/f89acfb9d7a60eaefa540af05dbb37cfa4148226/engine/engine/wscript#L35-L52

**Comment by AGulev** [comment]
> maybe it's possible to do something like that but on Extender?
> 
> https://github.com/defold/defold/blob/f89acfb9d7a60eaefa540af05dbb37cfa4148226/engine/engine/wscript#L35-L52

@britzl here we have it on extender side https://github.com/defold/defold/blob/59fbd647719377a557234d52eb7c2c559bc7392b/share/extender/build.yml#L13-L68

**Comment by subsoap** [comment]
Was this cut?

**Comment by britzl** [comment]
We are working on a better solution. Should be released in 1.3.2. The work with Spine has unlocked build/compile extensions for bob and the editor and we will use this in combination with a runtime API to let developers decide on how they wish to encrypt their resources. More info to come soon.

**Pull request closed without merging**